cmd/main.go

```diff
@@ -286,6 +291,24 @@ func setupReconcilers(mgr ctrl.Manager) {
 		setupLog.Error(err, "unable to create controller", "controller", "Healthcheck")
 		os.Exit(1)
 	}
+
+	if watchConfigSecretChanges {
```

Comment:
I like the idea of watching secret changes, but I don’t like that it requires a separate controller. Hash of both object can be combined, and stored under the same annotation.

Controller can establish watch on secrets with object mapper, which will trigger reconcile for all providers where the secret is referenced.

Reply:
Thanks or the review.
I have updated the code to watch for secret changes and trigger a reconciliation of each Provider using them

Let me know what you think